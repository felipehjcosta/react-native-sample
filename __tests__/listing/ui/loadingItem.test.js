import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import LoadingItem from '../../../src/listing/ui/loadingItem'
import ReactTestRenderer from 'react-test-renderer'

Enzyme.configure({adapter: new Adapter()})

describe('LoadingItem Component', () => {
  beforeEach(() => {
    jest.useFakeTimers()
    jest.mock('react-native-shimmer-placeholder', () => 'View')
  })

  it('renders correctly', () => {
    const wrapper = shallow(<LoadingItem />)

    const tree = ReactTestRenderer.create(wrapper).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
