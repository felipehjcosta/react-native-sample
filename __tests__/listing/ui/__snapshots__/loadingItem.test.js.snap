// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingItem Component renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "height": 15,
          "width": 200,
        },
        Object {
          "overflow": "hidden",
        },
        Object {
          "height": 448.2,
          "marginBottom": 8,
          "marginLeft": 16,
          "marginTop": 16,
          "width": 718,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ebebeb",
          "flex": 1,
        }
      }
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "flex": 1,
            "transform": Array [
              Object {
                "translateX": -200,
              },
            ],
          }
        }
      >
        <BVLinearGradient
          colors={
            Array [
              4293651435,
              4291151301,
              4293651435,
            ]
          }
          endPoint={
            Object {
              "x": 2,
              "y": 0.5,
            }
          }
          locations={
            Array [
              0.3,
              0.5,
              0.7,
            ]
          }
          startPoint={
            Object {
              "x": -1,
              "y": 0.5,
            }
          }
          style={
            Object {
              "flex": 1,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "height": 0,
            "width": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;
